obspy/io/seiscomp/tests/test_event.py

```diff
@@ -88,6 +90,8 @@ def cmp_read_xslt_file(self, sc3ml_file, quakeml_file, validate=True):
                 if validate:
                     self.assertTrue(_validate_quakeml(tf.name))
                 filepath_cmp = os.path.join(self.path, quakeml_file)
+                if debug and not filecmp.cmp(filepath_cmp, tf.name):
+                    shutil.copyfile(tf.name, filepath_cmp + "_failed")
```

Comment:
It's indeed pretty annoying to debug the file comparison. However if this file is written on the disk, it should also be the case for the other tests. Either we write them all or none.

We can perhaps also keep this mechanism with the debug parameter always false. The failed files will be generated only if a developper manually sets it to true. This solution is not perfect since it adds some unreachable code… I don't know what would be best.